### ISSUE TITLE
Citation scheme for Plato's Definitiones

### ISSUE THREAD

**Issue by sonofmun** [OpenGreekAndLatin/First1KGreek#840]
This is tlg0059.tlg037.First1K-grc1.xml
There is no citation scheme right now, but it appears to be nicely broken down into `<p>` elements, which may actually be better represented as `<l>`. Who has an idea as to how this text is normally cited? @lmuellner @Eumaeus @gregorycrane

**Comment by lmuellner** [comment]
The reference system is Stephanus pages as for all the other works of Plato — in the XML the numbers are marked up as <pb n=“411c”>, and the standard way to get to a Plato citation is with that number and in addition a line number, like 411c5, where the line breaks are on the printed page. So the <p> elements are not the way to go — I’m not sure what has been done previously with other works of Plato, but they should be your guide here.

**Comment by lmuellner** [comment]
This problem needs to be resolved -- it seems to have been resolved for Aristotle, but by using a competing logical system along with the Bekker page numbers. I'm not sure, @sonofmun and @gregorycrane, whether you've worked out a solution for this stuff or not?

**Comment by lcerrato** [comment]
There is current work being done on Plato where I have created a "perseus" section that follows Stephanus pages as best as possible in accordance with breaks.
This was something we discussed a while ago within Perseus re Aristotle, Plato and similar works based on specific print page editions.
I try to start the "perseus" section so that the new Stephanus page appears within that section whenever the break is not clean. So if the Stephanus page break occurs in the midst of a phrase, I break at the prior stop. 

```
<milestone unit="section" resp="Stephanus" n="18e"/> 
τοὺς πάλαι οὓς ἐγὼ λέγω, καὶ οἰήθητε δεῖν πρὸς ἐκείνους  
πρῶτόν με ἀπολογήσασθαι· καὶ γὰρ ὑμεῖς ἐκείνων πρότερον 
ἠκούσατε κατηγορούντων  καὶ πολὺ μᾶλλον ἢ τῶνδε τῶν ὕστερον.</p></div>
<div type="textpart" subtype="section" resp="perseus" n="19">
<p><milestone unit="para" ed="P"/>εἶεν·  ἀπολογητέον δή, ὦ ἄνδρες Ἀθηναῖοι, 
καὶ ἐπιχειρητέον 
<milestone n="19"  unit="page"/>
<milestone unit="section" resp="Stephanus" n="19a"/> ὑμῶν ἐξελέσθαι
```

Any of the Plato issues in canonical-greekLit should provide an example.
https://github.com/PerseusDL/canonical-greekLit/issues?utf8=✓&q=tlg0059

I am open to other ideas. This was a solution that enabled the conversion to proceed.

**Comment by lmuellner** [comment]
Hi Lisa, this is very helpful — and solves the problem in a reasonable if somewhat arbitrary way.  but the only Plato text in the repository at the moment is the one that elicited the question from Matt and my comment, the OPOI/Deinitiones, and its markup is very different from what you have below. It marks up the dictionary-style headers on Burnet’s edition (so if a given page starts in the middle of 411c, it encodes 411c — that’s useless. It encodes the a, b, c, etc of individual Steph. pages as marginal notes, and the actual starting Steph. page number also as a marginal note. We can figure these things out, but it will not be a small job to reencode these files, that’s for sure! Here’s a question: is it worth it to reencode what are now “marginal notes” as milestone elements? They certainly are that, and it would be good to have some consistency across these files, don’t you think? Best, Lenny

**Comment by sonofmun** [comment]
For Aristotle, when I have added the citation scheme myself, as @lmuellner says, I used the citation scheme from the edition instead of (what I assume are) the Bekker pages. If those do not exist here, then there needs to be something else done. What I have typically done is used the numbered `<pb>` or `<lb>` elements as a guide and tried to find where the break would have been on that line, i.e., searching for the ends of sentences or clauses marked by some sort of punctuation (typically using a regex). But since these are page numbers, they do not necessarily represent any sort of logical unit. I think marking them up as milestones, that will show up in the reading environment and allow people to use a traditional Stephanus citation if they wish may be our best bet here, which is what I understand Lisa to be talking about. My suggestion: use the `<p>` elements as our "Perseus" units, as Lisa called them, while marking up the Stephanus pages as milestones that will appear in the text. That way, people can use the CTS URN as a machine-actionable way to get to the part they are citing while, at the same time, mentioning the Stephanus numbers for people who want to check a print edition.

**Comment by lcerrato** [comment]
Yes, @sonofmun we need the Stephanus pages marked as divs for any editor that wants to reuse the texts this way and, importantly, find things. Greg would have preferred we just use the Stephanus numbers as the cts default as he said, it's a known, established reference scheme, however imperfect but we can't because of tree banking breaks.  

The Oxford works in Perseus we use for Plato have no other edition specific citation scheme embedded — even the paragraphs are inconsistently marked and there are no modern page numbers. Plus, they tend to vary greatly in size and, at least in this transitional phase, the larger chunks make more sense to me based on user needs. Again, another editor may have made a different decision. I was under pressure to produce this for classroom work and had to move forward.

All of my work is in canonical-greekLit — some not checked in yet — but some online.

This is a really tiny work — less than 6 Stephanus pages — I would be happy to mock up something along the lines of what I did for other Plato if that would help illustrate my approach.

**Comment by gregorycrane** [comment]
I am working my way through the Republic and that has chapter breaks in the Adams edition: e.g., chapter 9 marked on https://archive.org/stream/republicplato00adamgoog#page/n104/mode/2up

Is it worth using these? Or would it be confusing to use book/chapter for the Republic and rationalized stephanus page break for the rest?

**Comment by lcerrato** [comment]
In the case of Cicero, where this is a common issue, I tried to use whatever scheme is noted in reference works and avoided single-edition/editor style reference schemes. So for example, where Loeb had it's own scheme, I defaulted to the one mentioned in the OLD front matter. I didn't remove the Loeb chapters/sections.

It looks like book/stephanus page is the standard for the Republic but as the editor you may certainly default to the Adams numbering for the Adams edition if it makes sense to you.

**Comment by gregorycrane** [comment]
no one ever used the chapters that i see in adams (even adams doesn't!) but they are there. still, i think that moving stephanus pages to sentence breaks would be the consistent thing to do.
